### PR TITLE
Upgrade Netty to 4.1.36

### PR DESCRIPTION
**Pull request by dlogothetis**
- Upgrades Netty to 4.1.36
- Auto-reading used to be an option. Experiments showed that disabling it not reliable with the new version, so we'll keep it permanently enabled.

https://issues.apache.org/jira/browse/GIRAPH-1228

Tests:
- Unit tests
- Ran a number of large jobs and performance is similar.

### CODE REVIEW

File: giraph-core/src/main/java/org/apache/giraph/comm/netty/NettyClient.java [hunk 1]
```diff
@@ -335,6 +333,10 @@ protected void initChannel(SocketChannel ch) throws Exception {
             if (conf.authenticate()) {
               LOG.info("Using Netty with authentication.");
 
+              PipelineUtils.addLastWithExecutorCheck("flushConsolidation",
+                new FlushConsolidationHandler(FlushConsolidationHandler
+                  .DEFAULT_EXPLICIT_FLUSH_AFTER_FLUSHES, true),
+                handlerToUseExecutionGroup, executionGroup, ch);
```
Comment (aanchal204) [inline giraph-core/src/main/java/org/apache/giraph/comm/netty/NettyClient.java:339]: Why was this needed to be added?

**Review by aanchal204 — APPROVED** [review]
Everything looks good to me!

File: giraph-core/src/main/java/org/apache/giraph/comm/netty/NettyClient.java (hunk 1 above)
Comment (dlogothetis) [reply giraph-core/src/main/java/org/apache/giraph/comm/netty/NettyClient.java:339]: Looking around on netty related posts, it seems this is good practice as it batches small flushes into bigger ones for efficiency.
